File: 0080_remove_duplicates_from_sorted_array_II/remove_duplicates.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int removeDuplicates(int[] nums) {
        int j = 0;
        for (int i: nums) {
            if (j < 2 || nums[j - 2] < i) {
                nums[j++] = i;
            }
        }
        return j;
    }
}

File: 0080_remove_duplicates_from_sorted_array_II/remove_duplicates.cpp
#include <iostream>
#include <vector>

using namespace std;

/**
 * Time: O(n)
 * Space: O(1)
 */
int remove_duplicates(vector<int> &nums)
{
    int i = 0;
    for (auto e : nums)
        if (i < 2 || nums[i - 2] < e)
            nums[i++] = e;
    return i;
}